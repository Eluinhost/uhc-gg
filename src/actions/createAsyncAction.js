import { createAction } from 'redux-actions';
import { always } from 'ramda';

const emptyPayload = always({});

export const createAsyncAction = (name, options = {}) => {
  const trigger = createAction(
    name,
    options.triggered || emptyPayload,
    options.triggeredMeta || emptyPayload
  );

  trigger.STARTED = createAction(
    `${name}.STARTED`,
    options.started || emptyPayload,
    options.startedMeta || emptyPayload
  );
  trigger.COMPLETED = createAction(
    `${name}.COMPLETED`,
    options.completed || emptyPayload,
    options.completedMeta || emptyPayload
  );

  return trigger;
};
