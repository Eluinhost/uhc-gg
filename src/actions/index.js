import { createAction } from 'redux-actions';
import { createAsyncAction } from './createAsyncAction';

export const SET_DRAWER_OPEN = createAction('SET_DRAWER_OPEN', isOpen => ({
  isOpen,
}));

export const UPDATE_UPCOMING_MATCHES = createAsyncAction(
  'UPDATE_UPCOMING_MATCHES',
  {
    completed: matches => ({ matches }),
  }
);
