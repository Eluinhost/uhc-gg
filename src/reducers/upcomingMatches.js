import { UPDATE_UPCOMING_MATCHES } from '../actions';
import { handleActions } from 'redux-actions';

export const upcomingMatches = handleActions(
  {
    [UPDATE_UPCOMING_MATCHES.STARTED]: state => ({
      isFetching: true,
      error: null,
      matches: state.matches,
    }),
    [UPDATE_UPCOMING_MATCHES.COMPLETED]: {
      next: (state, action) => ({
        isFetching: false,
        error: null,
        matches: action.payload.matches,
      }),
      throw: (state, action) => ({
        isFetching: false,
        error: action.payload,
        matches: state.matches,
      }),
    },
  },
  {
    isFetching: false,
    error: null,
    matches: [],
  }
);
