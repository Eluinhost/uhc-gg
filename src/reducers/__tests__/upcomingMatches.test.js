import { UPDATE_UPCOMING_MATCHES } from '../../actions';
import { upcomingMatches } from '../upcomingMatches';

describe('upcomingMatches reducer', () => {
  test('UPDATE_UPCOMING_MATCHES.STARTED', () => {
    const pre = {
      isFetching: false,
      error: new Error('test'),
      matches: [1, 2, 3],
    };

    const post = upcomingMatches(pre, UPDATE_UPCOMING_MATCHES.STARTED());

    expect(post).toEqual({
      isFetching: true,
      error: null,
      matches: [1, 2, 3],
    });
    expect(post).toMatchSnapshot();
  });

  test('UPDATE_UPCOMING_MATCHES.COMPLETED success', () => {
    const pre = {
      isFetching: true,
      error: new Error('test'),
      matches: [1, 2, 3],
    };

    const post = upcomingMatches(
      pre,
      UPDATE_UPCOMING_MATCHES.COMPLETED([3, 4, 5])
    );

    expect(post).toEqual({
      isFetching: false,
      error: null,
      matches: [3, 4, 5],
    });
    expect(post).toMatchSnapshot();
  });

  test('UPDATE_UPCOMING_MATCHES.COMPLETED failure', () => {
    const pre = {
      isFetching: true,
      error: null,
      matches: [1, 2, 3],
    };

    const post = upcomingMatches(
      pre,
      UPDATE_UPCOMING_MATCHES.COMPLETED(new Error('test'))
    );

    expect(post).toEqual({
      isFetching: false,
      error: new Error('test'),
      matches: [1, 2, 3],
    });
    expect(post).toMatchSnapshot();
  });
});
