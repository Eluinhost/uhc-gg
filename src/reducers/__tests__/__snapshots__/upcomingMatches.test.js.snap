// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`upcomingMatches reducer UPDATE_UPCOMING_MATCHES.COMPLETED failure 1`] = `
Object {
  "error": [Error: test],
  "isFetching": false,
  "matches": Array [
    1,
    2,
    3,
  ],
}
`;

exports[`upcomingMatches reducer UPDATE_UPCOMING_MATCHES.COMPLETED success 1`] = `
Object {
  "error": null,
  "isFetching": false,
  "matches": Array [
    3,
    4,
    5,
  ],
}
`;

exports[`upcomingMatches reducer UPDATE_UPCOMING_MATCHES.STARTED 1`] = `
Object {
  "error": null,
  "isFetching": true,
  "matches": Array [
    1,
    2,
    3,
  ],
}
`;
